### PR TITLE
Improve type label support

### PR DESCRIPTION
**Pull request by jrgissing**
**Summary**

add direct type label support for commands/fixes/computes that have 'type' arguments

- [x] create_atoms command
- [x] fix deposit
- [x] fix mol/swap
- [x] fix atom/swap
- [x] group command
- [ ] fix shake/rattle
  - [x] implemented but not committed. type labels could pretty easily interfere with 'a', 'b', etc. keywords
- [x] fix gcmc
- [x] fix widom
- [x] delete_bonds command
- [x] fix bond/create/break
- [x] fix adapt + adapt/fep
  - [x] add option to feed a type label into utils::bounds. returns nlo = nhi = label2type(type_label)
  - [x] I < J requirement relaxed for non-asterisk syntax
- [x] compute rdf
  - [x] could technically break there was a type label called 'cutoff'
- [ ] pair_style tracker
  - [x] not sure if can work, because commas used in lists
- [x] fix charge/regulation
- [ ] variable command (type==H)
  - [x] rely on label2type(kind,label)
- [x] Dump Output Support
  - [x] dump_modify 'types' for xyz
  - [x] 'typelabel' keyword for dump custom
- [ ] compute property/local - will require refactor of communication routines, suggest a separate PR
- [ ] others?

Existing/prior support includes:
- [x] data file/molecule templates
- [x] label2type function
- [x] set command
- [x] coeff commands
- [x] mass command

**Related Issue(s)**

none

**Author(s)**

Jake

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

yes

**Implementation Notes**

new string output (type labels) to dump files not added in a very extensible manner. can refactor if you think more string fields will be added to dump output

~~currently, 'dump_modify types numeric' (only relevant to dump xyz) does not revert a previous 'dump_modify types labels' command~~

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

File: doc/src/create_atoms.rst [hunk 1]
```diff
@@ -74,7 +74,13 @@ Examples
 .. code-block:: LAMMPS
 
    create_atoms 1 box
-   create_atoms 3 region regsphere basis 2 3
```
Comment (akohlmey) [inline doc/src/create_atoms.rst:77]: Please don't delete examples for numerical inputs

**Comment by stanmoore1** [comment]
Still to do: add support for dump files

**Comment by jrgissing** [comment]
> Still to do: add support for dump files

dump file support is complete

**Comment by akohlmey** [comment]
@sjplimp @rbberger @athomps @jtclemm This is now flagged as ready for review and needs to be reviewed.

**Comment by akohlmey** [comment]
@jrgissing It would be great if there would be some tests in the unittest tree added for the new functionality.

**Review by sjplimp — APPROVED** [review]

**Comment by jrgissing** [comment]
> @jrgissing It would be great if there would be some tests in the unittest tree added for the new functionality.

Good idea. I am in a bit of a crunch for the next week and a half, but I will add some unit tests after that

**Comment by akohlmey** [comment]
> > @jrgissing It would be great if there would be some tests in the unittest tree added for the new functionality.
> 
> Good idea. I am in a bit of a crunch for the next week and a half, but I will add some unit tests after that

Ok. Then I'll merge this "as is" now, so it will be in the next feature release and the tests can be in a new pull requests. I may be able to do a few of them on the side (the simple ones)

**Pull request merged**